### PR TITLE
Support setting SO_LINGER on TcpSocket

### PR DESCRIPTION
**Pull request by zekisherif**
Because tokio `TcpStream` and mio `TcpStream` now use `std::net::TcpStream` as the underlying stream, there isn't a way to set linger options.

I'm aware of #1381 and the dislike of duplicating features. This is the last functionality I need.

The plan is to convert the stream by calling `as_raw_fd()` and set the linger option.

### CODE REVIEW

File: tests/tcp_socket.rs [hunk 1]
```diff
@@ -58,6 +59,19 @@ fn get_localaddr() {
...
 }
 
+#[test]
+fn set_linger() {
+    let addr = "127.0.0.1:0".parse().unwrap();
+
+    let socket = TcpSocket::new_v4().unwrap();
+    socket.set_linger(Some(Duration::from_secs(1))).unwrap();
+    assert_eq!(socket.get_linger().unwrap().unwrap().as_secs(), 1);
+
```
Comment (Thomasdezeeuw) [inline tests/tcp_socket.rs:69]: Can you also test `set_ling(None)`?

**Review by Thomasdezeeuw — APPROVED** [review]
LGTM if you can fix the CI and the small change to the test.

**Comment by Thomasdezeeuw** [comment]
Thanks @zekisherif.

**Comment by zekisherif** [comment]
Just realized that the commit message should really be "Support getting SO_LINGER value".

The setting of the linger value is meant for tokio TcpStream not mio.

**Comment by Thomasdezeeuw** [comment]
I missed it too, oh well.. it's committed now.
